src/de/impls/visitor/enum.zig

```diff
@@ -19,15 +20,57 @@ pub fn Visitor(comptime Enum: type) type {
         const Value = Enum;
 
         fn visitEnum(_: Self, _: ?std.mem.Allocator, comptime Deserializer: type, input: anytype) Deserializer.Error!Value {
-            return input;
+            return visitString(@tagName(input));
         }
 
         fn visitInt(_: Self, _: ?std.mem.Allocator, comptime Deserializer: type, input: anytype) Deserializer.Error!Value {
-            return std.meta.intToEnum(Value, input) catch error.InvalidValue;
+            const fields = std.meta.fields(Value);
+            const attributes = comptime getAttributes(Value, Deserializer);
+            const result = std.meta.intToEnum(Value, input) catch error.InvalidValue;
+            const resulta = try result;
+
+            inline for (fields) |field| {
+                if (std.mem.eql(u8, field.name, @tagName(resulta))) {
```

Comment:
We can replace this with `std.meta.isTag`, which is O(1).

```suggestion
                if (std.meta.isTag(result, field.name)) {
```

Reply:
Oh, and at the end of this `isTag` block, we should have a `break` so that we can skip irrelevant fields.